### ISSUE TITLE
[Low prio] Configurable “slur” filter

### ISSUE THREAD

**Issue by kromonos** [jointakahe/takahe#289]
For later implementation, when filtering is implemented, it would be really loveable, to get configurable “slur” filter, for server wide moderation helpings.  
I know this from Lemmy, where the administrator can configure a regex string to filter things out and prevent it from being posted. Lemmy had it hard coded at first, but then they changed it for configuration. The default one is `(fag(g|got|tard)?\b|cock\s?sucker(s|ing)?|(n|i){1,32}((g{2,32}|q){1,32}|[gq]{2,32})[e3r]{1,32}(s|z)?|mudslime?s?|kikes?|\bspi(c|k)s?\b|\bchinks?|gooks?|bitch(es|ing|y)?|whor(es?|ing)|\btr(a|@)nn?(y|ies?)|\b(b|re|r)tard(ed)?s?)`

Many don't like such filters and there have been many heated debates about them, but I find them very helpful for moderating, especially if you have a small team.

**Comment by andrewgodwin** [comment]
Yes, I would consider this more of an "automod", but the ability to automatically report or block things with certain words or URLs in them would be good.

**Comment by kromonos** [comment]
Oh, yeah. AutoMod sounds way better 👍

**Comment by TkTech** [comment]
This is another spot where it would shine as a plugin, just like these are usually implemented in WordPress and friends so it can be shared among peers, use an API, or filter differently (like for RTL languages)
